### ISSUE TITLE
[13.0] POS / Inventory issue with picking after upgrade from V12

### ISSUE THREAD

**Issue by E-HO** [OCA/OpenUpgrade#3188]
Impacted versions: 13.0

Steps to reproduce: Migrate a db from v12 with POS / Stock (Inventory)

Current behavior: POS is not working anymore (sales are not saved), pos cannot be closed properly

Hi,

**After an upgrade from V12, encounter issues mainly with the existing POS :**

    On POS GUI, when trying to close the POS, there is a red box with an error message : "You must be connected to the internet to save your changes. Orders that where not synced before will be synced next time you close an order while connected to the internet or when you close the session." and "Something went wrong ! Reference must be unique per company!".

**In logs :**

    odoo.sql_db: bad query: INSERT INTO "stock_picking" (..."name"...) VALUES (..."/"...)
    ERROR: duplicate key value violates unique constraint "stock_picking_name_uniq"
    DETAIL: Key (name, company_id)=(/, 1) already exists.

Note that the error is that the name is no more a POS order name like "POS/xyz" but just a unique "/".

Trying to archive the POS and to create a new one don't change anything.

Thanks,

**Comment by pedrobaeza** [comment]
The picking sequence is taking an already existing picking number. Change that.

**Comment by E-HO** [comment]
In this case, error shouldn't be at least "POS_Prefix/" ?

Tried to add one supplementary thousand to each sequences (pos.order.line , pos.order, pos.session), but it's unfortunately still the same result.

**Comment by pedrobaeza** [comment]
Check the PoS picking sequence.

**Comment by E-HO** [comment]
This entry was completely deleted, probably bu the update who was also trying to delete the whole pos.

But even with re-entering it with proper values, don't give better results. Detail : in the SQL statement in the log, the pos sequence is correctly set in "origin" (while still "/" as name).

**Comment by pedrobaeza** [comment]
Sorry, but I don't think that's a problem in OU, but on your initial DB state.

**Comment by E-HO** [comment]
Not sure of all the reasons, but seems I found a way to make it fully working.

Because the issue was linked to "picking", I looked at the picking options in the "POS Settings / Inventory / Operation Type". So I finally created and assigned a new value for this, without using anymore the old existing one ... and things came back to life.

![image](https://user-images.githubusercontent.com/16776195/160023187-447daf96-0b36-4045-b3d3-2027ba83ce23.png)
